src/Symfony/Component/DependencyInjection/Tests/Compiler/ResolveDefinitionTemplatesPassTest.php

```diff
@@ -139,6 +139,25 @@ public function testProcessDoesNotCopyDecoratedService()
         $this->assertNull($def->getDecoratedService());
     }
 
+    public function testProcessDoesNotDropShared()
+    {
+        $container = new ContainerBuilder();
+
+        $container
+            ->register('parent')
+        ;
+
+        $container
+            ->setDefinition('child', new DefinitionDecorator('parent'))
+            ->setShared(false)
```

Comment:
IMO you should put this `setShared(false)` in the definition of the parent service in order to test if we pass the same shared value to their childrens.